Fix sandbox violation caused by calling ln in /sbin instead of $(DESTDIR)/sbin

Also fixed an inverted race condition when installing with MAKEOPTS="-j1"

jer found this:
  install-root_sbinPROGRAMS is called from install-data-am but
  install-exec-hook is called from install-exec-am

So moving the failing ln call into install-data-hook for now...

--- cifs-utils-6.10/Makefile.am
+++ cifs-utils-6.10/Makefile.am
@@ -118,11 +118,9 @@
 
 SUBDIRS = contrib
 
-install-exec-hook:
-	(cd $(ROOTSBINDIR) && ln -sf mount.cifs mount.smb3)
-
 install-data-hook:
-	(cd $(man8dir) && ln -sf mount.cifs.8 mount.smb3.8)
+	(cd $(DESTDIR)$(ROOTSBINDIR) && ln -sf mount.cifs mount.smb3)
+	(cd $(DESTDIR)$(man8dir) && ln -sf mount.cifs.8 mount.smb3.8)
 
 uninstall-hook:
 	(cd $(ROOTSBINDIR) && rm -f $(ROOTSBINDIR)/mount.smb3)
